pkg/apis/arn.go

```diff
@@ -28,19 +28,23 @@ import (
 type ARN arn.ARN
 
 var (
-	_ fmt.Stringer             = (*ARN)(nil)
-	_ encoding.TextMarshaler   = (*ARN)(nil)
-	_ encoding.TextUnmarshaler = (*ARN)(nil)
+	_ fmt.Stringer     = (*ARN)(nil)
+	_ json.Marshaler   = (*ARN)(nil)
+	_ json.Unmarshaler = (*ARN)(nil)
 )
 
 // String implements the fmt.Stringer interface.
 func (a ARN) String() string {
 	return arn.ARN(a).String()
 }
 
-// UnmarshalText implements encoding.TextUnmarshaler.
-func (a *ARN) UnmarshalText(data []byte) error {
-	dataStr := string(data)
+// UnmarshalJSON implements json.Unmarshaler.
+func (a *ARN) UnmarshalJSON(data []byte) error {
+	var dataStr string
+	if err := json.Unmarshal(data, &dataStr); err != nil {
+		return err
+	}
```

Comment:
If the data passed to that method is a valid JSON string, it will look like `"arn:aws:..."` (quotes included). Passing those bytes through `json.Unmarshal` is safer than trying to manually trim the surrounding double-quotes.